Add fields for Worldwide Organisations

brucebolt opened the pull request:
In https://github.com/alphagov/whitehall/pull/7191, we have added additional fields to be presented as part of Worldwide Organisation content items.  This adds those additional fields to the schema.

Ultimately, this work will enable us to get the content for Worldwide Organisations into the content item, meaning the rendering can be moved out of Whitehall.

[Trello card](https://trello.com/c/3r6TFSCF)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

kevindew commented on content_schemas/formats/shared/definitions/_logo.jsonnet:
```diff
@@ -0,0 +1,32 @@
+{
```
Shall we call this file organisation logo? I'm wondering if logo is a bit too generic as it's definitely constrained to a few scenarios.

brucebolt replied:
Good idea, I'll rename this.

kevindew commented on content_schemas/formats/shared/definitions/_logo.jsonnet:
```diff
@@ -0,0 +1,32 @@
+{
+  type: "object",
```
Are any of the properties here required?

brucebolt replied:
Looks like the title is required, but the crest and image are optional.  I'll add this as required.

kevindew commented on content_schemas/formats/shared/definitions/_logo.jsonnet:
```diff
@@ -0,0 +1,32 @@
+    formatted_title: {
+      type: "string",
+    },
+    crest: {
```
This looks a bit complex with allowing string and null and then having null in the enum. Though this is just a move I wonder if we can simplify it.

I wonder if it'd be better to require a string for this and then to just not set the property all together if this isn't defined (i.e. capitalise on it not being a required property)

brucebolt replied:
I've removed both `null` from this property as it's optional anyway.

kevindew commented on content_schemas/formats/shared/definitions/_logo.jsonnet:
```diff
@@ -0,0 +1,32 @@
+{
+  type: "object",
+  properties: {
```
Should this also have additional properties set to false?

brucebolt replied:
Yes, it should.  Have added that in.

kevindew commented on content_schemas/formats/worldwide_organisation.jsonnet:
```diff
@@ -4,6 +4,26 @@
+              title: {
+                type: "string",
+              },
+              href: {
```
I'm probably drifting into the optimistic here, but are these corporate information pages available via the links hash? I'm a bit concerned that title and href are rather delicate attributes that have a tight coupling.

I'm wondering if this could be done entirely through expanded links, assuming title is not customised and if title is customised here whether the href could be a content_id instead so the href can be pulled out of expanded links

brucebolt replied:
Unfortunately the title comes from Whitehall arbitrarily defining what to include here (sometimes it doesn't match the name of the document, being pulled in from the locale files).  It's the same for organisation pages too.

kevindew replied:
Sure, that seems fine, would the href be able to be a content_id then so the document can be matched from the links rather than a hardcoded URL?

brucebolt replied:
That should be possible.  I'll have to add the pages in as links first.

brucebolt replied:
I've added these in as links and replaced the `href` in the details hash with a content ID.

kevindew reviewed (commented):
Looks good 👍  just a few suggestions.

kevindew commented:
Oh and for future, when editing schemas it's a bit easier to review if you have a single commit at the end that regenerates the schemas, otherwise it's quite hard to determine what are your changes amongst the regeneration ones (aside, I dream of us not committing the generated schemas and them being a deploy time artefact)

brucebolt commented:
Thanks @kevindew, looking at addressing your comments now.

> when editing schemas it's a bit easier to review if you have a single commit at the end that regenerates the schemas

Noted.  I'll update this so it's easier to re-review.

kevindew commented on content_schemas/formats/shared/definitions/_organisation_logo.jsonnet:
```diff
@@ -10,7 +10,6 @@
       type: [
         "string",
-        "null",
       ],
```
```suggestion
      type: "string"
```

minor, you don't need to define this in an array.

brucebolt replied:
Thanks.  I'll fix this.

kevindew reviewed (approved):
Looks great 👍 thanks for the changes. Just a minor syntactic issue.

Do we know if the existing organisation type is sending any data with a crest of null? I imagine we need to ensure Whitehall is omitting the fields when the data doesn't exist?

brucebolt commented:
> Do we know if the existing organisation type is sending any data with a crest of null? I imagine we need to ensure Whitehall is omitting the fields when the data doesn't exist?

That shouldn't be happening as [we are using `compact` on the logo hash](https://github.com/alphagov/whitehall/blob/4aec232784a537e581b4962ba5c49f9b0b5366ea/app/presenters/publishing_api/organisation_presenter.rb#L77-L81) so the nil values are all being removed.

kevindew commented:
Great stuff 👍 merge away
